File: lua/prime.lua
#!/usr/bin/lua

min=tonumber(arg[1])
max=tonumber(arg[2])

print("Calculam de la " .. string.format("%.0f",min) .. " la " .. string.format("%.0f",max))

for i=min,max,1 do
	prim=true
	for j=2,math.sqrt(i),1 do
		if math.fmod(i,j)==0 then
			prim=false
			break
		end
	end
	if prim then
		print(string.format("%.0f",i))
	end
end

File: go/src/prime.go
package main

import "fmt"
import "os"
import "strconv"
import "math"

func is_prime(i float64) bool{
	var j float64
	prim:=true

	for j=2;j<=math.Sqrt(i);j++{
		if math.Mod(i,j)==0{
			prim=false
			break
		}
	}
	if prim==true{
		return true
	}
	return false
}

func main() {
	var min, max float64
	min, _ = strconv.ParseFloat(os.Args[1],64)
	max, _ = strconv.ParseFloat(os.Args[2],64)
	var i float64
	for i=min;i<=max;i++{
		if is_prime(i)==true{
			fmt.Printf("%.0f\n", i)
		}
	}
}

